test/scripts/generate-handlers-json.spec.ts

```diff
@@ -0,0 +1,55 @@
+import getDriver, { Driver } from "./generate-handlers-json.driver";
+
+describe("generate-events", () => {
+  let driver: Driver;
+  beforeEach(() => {
+    driver = getDriver();
+  });
+
+  it("should return the base element event handlers", () => {
+    driver.given.mockDeclarationsFile();
+
+    const elementHandlers = driver.get.componentHandlersNames("Element");
+
+    expect(elementHandlers).toEqual([
+      "onMouseIn",
+      "onMouseOut",
+      "onViewportEnter",
+      "onViewportLeave"
+    ]);
+  });
+
+  it("should search recursivley for handlers", () => {
+    driver.given.mockDeclarationsFile();
+
+    const elementHandlers = driver.get.componentHandlersNames("Chatbox");
+
+    expect(elementHandlers).toEqual([
+      "onMouseIn",
+      "onMouseOut",
+      "onViewportEnter",
+      "onViewportLeave"
+    ]);
+  });
+
+  it("should return an object with all the handler details", () => {
+    driver.given.mockDeclarationsFile();
```

Comment:
`beforeEach`?

Reply:
I think I like it better outside the before each, so it's follows the pattern: given -> when -> expect